File: web/tests/reference/bst/insert-into-a-binary-search-tree.js
function insertIntoBST(root, val) {
  if (root === null) return new TreeNode(val);
  let node = root;
  for (;;) {
    if (val < node.val) {
      if (node.left === null) {
        node.left = new TreeNode(val);
        return root;
      }
      node = node.left;
    } else {
      if (node.right === null) {
        node.right = new TreeNode(val);
        return root;
      }
      node = node.right;
    }
  }
}

File: web/tests/reference/bst/insert-into-a-binary-search-tree.py
def insert_into_bst(root, val):
    if root is None:
        return TreeNode(val)
    node = root
    while True:
        if val < node.val:
            if node.left is None:
                node.left = TreeNode(val)
                return root
            node = node.left
        else:
            if node.right is None:
                node.right = TreeNode(val)
                return root
            node = node.right
